File: api/src/main/java/net/impactdev/gts/api/util/TriFunction.java
package net.impactdev.gts.api.util;

@FunctionalInterface
public interface TriFunction<T, U, V, R> {

	/**
	 * Applies this function to the given arguments.
	 *
	 * @param t the first function argument
	 * @param u the second function argument
	 * @return the function result
	 */
	R apply(T t, U u, V v);

}

File: api/src/main/kotlin/net/impactdev/gts/api/util/TriFunction.kt
package net.impactdev.gts.api.util

@FunctionalInterface
interface TriFunction<T, U, V, R> {
    /**
     * Applies this function to the given arguments.
     *
     * @param t the first function argument
     * @param u the second function argument
     * @return the function result
     */
    fun apply(t: T, u: U, v: V): R
}